assignment/scripts/function-practice.js

```diff
@@ -49,41 +65,110 @@ console.log( 'isPositive - should say false', isPositive(-3) );
 
 // 6. Function to return the _last_ item in an array. If the 
 //    array is empty, return `undefined`.
-function getLast( array ) {
-
+function getLast(array) {
+  if(array === []){
+    return `undefined`;
+  }
+  return array[array.length -1];
 }
 
+console.log('Last number of the array:', getLast([5, 4, 6, 10, 12]));
+
+//array.length=0
+
 // 7. Function to find a value in an array. Return true if the 
 //    value is found and false otherwise. Use a loop;
 //    DO NOT use Array.includes, Array.indexOf, or Array.find 
-function find( value, array ){
-  
+function find(value, array) {
+  for (let part of array) {
+        if (part === value) {
+          return true;
+        }
+      }
+      return false;
 }
 
+console.log('Is 1 in array?', find(1, [1, 2, 3]));
+console.log('Is Javi in the car?', find('Javi', ['Juan', 'Luis', 'Camila']));
+
 // ----------------------
 // Stretch Goals
 // ----------------------
 // 8. Function to check if a letter is the first letter in a 
 //    string. Return true if it is, and false otherwise
 function isFirstLetter(letter, string) {
-
+  for (let character of string){
+    if (character === letter){
+      return true;
+    }
+    return false;
```

Comment:
Since you have the `return false` inside the loop, this will work because the loop will only every run once before you return something. However, this alternate approach would be a lot simpler and lot easier to read:

```js
function isFirstLetter(letter, string) {
  return string[0] === letter;
}
```